clustermesh-apiserver/README.md

```diff
@@ -1,29 +1,85 @@
 # API server for Cilium ClusterMesh
 
+Cilium uses a clustermesh-apiserver in setups that do not use a KVstore as the
+identity allocation mode but use a CRD mode (e.g. external workloads).
```

Comment:
IMO this would be better worded as:

> Cilium uses a clustermesh-apiserver when multiple clusters are connected in clustermesh, or
when external workloads are connected in to the Cilium cluster.

Reply:
Wait a sec, but clustermesh-apiserver is not needed at all when cilium uses kvstore, isn't that true?

Reply:
because as far as I can see cilium-agent does exactly these operations when kvstore is used as identity allocation mode

Reply:
@jrajahalme any comments?

Reply:
`clustermesh-apiserver` is only deployed when needed for clustermesh or external workload support. If neither is used, then `clustermesh-apiserver` is never started.

kvstore use in general is an orthogonal issue.